### PR TITLE
source-easypost contribution from btkcodedev

### PR DESCRIPTION
**Pull request by btkcodedev**
## What

This PR adds a new connector for source Easypost (source-easypost).

Contributor-provided connector description:

Website: https://www.easypost.com/
API Docs: https://docs.easypost.com/docs/addresses
Auth Docs: https://docs.easypost.com/docs/authentication
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `username` | `string` | API Key. The API Key from your easypost settings |  |
| `start_date` | `string` | Start date.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| batches | id | DefaultPaginator | ✅ |  ✅  |
| addresses | id | DefaultPaginator | ✅ |  ❌  |
| trackers | id | DefaultPaginator | ✅ |  ✅  |
| metadata_carriers | name | No pagination | ✅ |  ❌  |
| end_shippers | id | DefaultPaginator | ✅ |  ❌  |
| webhooks | id | DefaultPaginator | ✅ |  ✅  |
| users_children | id | DefaultPaginator | ✅ |  ✅  |
| carrier_accounts | id | DefaultPaginator | ✅ |  ✅  |
| api_keys | id | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: docs/integrations/sources/easypost.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Easypost
+This directory contains the manifest-only connector for [`source-easypost`](https://www.easypost.com/).
+
+## Documentation reference:
+- Visit `https://docs.easypost.com/docs/addresses` for API documentation
```
Comment (natikgadzhi) [inline docs/integrations/sources/easypost.md:5]:
```suggestion
- Visit [`https://docs.easypost.com/docs/addresses`](https://docs.easypost.com/docs/addresses) for API documentation
```

File: docs/integrations/sources/easypost.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+| api_keys | id | DefaultPaginator | ✅ |  ❌  |
+
+## Changelog
+
+<details>
+  <summary>Expand to review</summary>
+
+| Version          | Date              | Pull Request | Subject        |
+|------------------|-------------------|--------------|----------------|
+| 0.0.1 | 2024-10-01 | [46287](https://github.com/airbytehq/airbyte/pull/46287) | Initial release by [@btkcodedev](https://github.com/btkcodedev) via Connector Builder |
```
Comment (natikgadzhi) [inline docs/integrations/sources/easypost.md:37]: Thank you @btkcodedev ;)

**Review by natikgadzhi — APPROVED** [review]
@btkcodedev to be useful, this connector needs "shipments" stream.

Most people who use posting APIs would need to see their actual shipments and their statuses over time.

I think we can merge this, but this needs a follow-up. I _think_ it's tolerable to add shipments with zero records read, but if you do it in a follow-up PR, then you will see the scope of the change.

**Comment by btkcodedev** [comment]
Yep, Shipments on its way

**Pull request merged**